src/libcore/macros.rs

```diff
@@ -15,7 +15,8 @@ macro_rules! panic {
         panic!("explicit panic")
     );
     ($msg:expr) => ({
-        static _MSG_FILE_LINE: (&'static str, &'static str, usize) = ($msg, file!(), line!());
+        static _MSG_FILE_LINE: (&'static str, &'static str, usize) =
```

Comment:
#19284 has more details on that approach.  Not much luck, so hoping someone familiar with this type of issue can shed some guidance.